docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
-transformation which changes the way a data stream or data set is partitioned is often called
-repartitioning.
+A partition is an independent subset of the overall [DataStream](#datastream). A DataStream is divided 
+into partitions by assigning each [record](#record) to one or more partitions via keys. Partitions of 
+DataStreams are consumed by [tasks](#task) during runtime. A transformation that changes the way a 
+DataStream is partitioned is often called repartitioning.
 
 #### Physical Graph
 
-A physical graph is the result of translating a [Logical Graph](#logical-graph) for execution in a
-distributed runtime. The nodes are [Tasks](#task) and the edges indicate input/output-relationships
-or [partitions](#partition) of data streams or data sets.
+A physical graph is the result of translating a [logical graph](#logical-graph) for execution in a
+distributed runtime. The nodes are [tasks](#task) and the edges indicate input/output relationships
+or [partitions](#partition) of DataStreams.
+
+#### POJO
+
+This is a composite data type and can be serialized with Flink's serializer. Flink recognizes a data 
+type as a POJO type (and allows “by-name” field referencing) if the following conditions are met:
+
+- the class is public and standalone (no non-static inner class)
+- the class has a public no-argument constructor
+- all non-static, non-transient fields in the class (and all superclasses) are either public (and 
+  non-final) or have public getter- and setter- methods that follow the Java naming conventions for 
+  getters and setters
+  
+Flink analyzes the structure of POJO types and can process POJOs more efficiently than general types.
+
+#### Process Functions
+
+This type of function combines event processing with timers and state and is the basis for creating 
+event-driven applications with Flink.
+
+#### Processing Time
+
+The time when a specific operator in your pipeline is processing the event. Computing analytics based 
+on processing time can cause inconsistencies and make it difficult to re-analyze historic data or test 
+new implementations.
+
+#### Queryable State 
+
+This is managed keyed (partitioned) state that can be accessed from outside of Flink during runtime.
+
+#### Raw State
+
+This is state that operators keep in their own data structures. When checkpointed, only a sequence of 
+bytes is written into the checkpoint and Flink knows nothing about the state’s data structures and will 
+see only the raw bytes.
+
+[Keyed state](#keyed-state) and [operator state](#operator-state) exist in two forms: [managed](#managed-state) and raw.
 
 #### Record
 
-Records are the constituent elements of a data set or data stream. [Operators](#operator) and
-[Functions](#Function) receive records as input and emit records as output.
+Records are the elements that make up a [DataStream](#datastream). [Operators](#operator) and [functions](#function) 
```

Comment:
I would use the [wikipedia description](https://en.wikipedia.org/wiki/Record_(computer_science)) for the first sentence, for example what about

```suggestion
A record is a collection of named fields of different data types. Streams of data are organized into records. [Operators](#operator) and [functions](#function) 
```